java-client/session/src/main/java/io/deephaven/client/impl/UnaryGrpcFuture.java

```diff
@@ -0,0 +1,93 @@
+package io.deephaven.client.impl;
+
+import io.grpc.stub.ClientCallStreamObserver;
+import io.grpc.stub.ClientResponseObserver;
+import io.grpc.stub.StreamObserver;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.function.BiConsumer;
+import java.util.function.Function;
+
+/**
+ * ClientResponseObserver implementation that can map results into a CompletableFuture. To keep generics under control,
+ * has two factory methods, {@link #of(Object, BiConsumer, Function)} and {@link #ignoreResponse(Object, BiConsumer)}.
+ * 
+ * @param <ReqT> The type of the gRPC request
+ * @param <RespT> The type of the gRPC response
+ * @param <V> The type of the result to make available in the future
+ */
+public class UnaryGrpcFuture<ReqT, RespT, V> implements ClientResponseObserver<ReqT, RespT> {
```

Comment:
Is there value in making this public? (I know we are very loose in other places w/ public...)

Reply:
My assumption was that just as DeephavenChannel is very-public (presumably to support the various use cases that java-client doesnt have implemented?), we want this tool available for downstream users to make unary calls easy to build with CFs.